python/lsst/ts/tcpip/one_client_read_loop_server.py

```diff
@@ -0,0 +1,143 @@
+        when a client connects or disconnects.
+        If the other end (client) closes the connection, it may take
+        ``monitor_connection_interval`` seconds or longer to notice.
+        The function receives one argument: this `OneClientServer`.
+    name : `str`, optional
+        Name used for log messages, e.g. "Commands" or "Telemetry".
+    **kwargs : `dict` [`str`, `typing.Any`]
+        Additional keyword arguments for `asyncio.start_server`,
+        beyond host and port.
+
+    Attributes
+    ----------
+    read_loop_task : `asyncio.Future`
+        A task that reads messages from the reader in a loop.
+    """
+
+    def __init__(
+        self,
+        host: str | None,
+        port: int | None,
+        log: logging.Logger,
+        connect_callback: ConnectCallbackType | None = None,
+        name: str = "",
+        **kwargs: typing.Any,
+    ) -> None:
+        self.read_loop_task: asyncio.Future = asyncio.Future()
+
+        super().__init__(
+            host=host,
+            port=port,
+            log=log,
+            connect_callback=connect_callback,
+            monitor_connection_interval=0,
+            name=name,
+            **kwargs,
+        )
+
+    async def call_connect_callback(self) -> None:
+        """A client has connected or disconnected."""
+        await super().call_connect_callback()
+        if self.connected:
+            self.log.info("Client connected.")
+            self.read_loop_task.cancel()
+            self.read_loop_task = asyncio.create_task(self.read_loop())
+        else:
+            self.log.info("Client disconnected.")
+
+    async def read_loop(self) -> None:
+        """Read incoming data and handle them.
+
+        The actual reading is deferred to the `read_and_dispatch` method and
+        needs to be implemented by subclasses.
+        """
+        try:
+            self.log.info(f"The read_loop begins connected? {self.connected}")
+            while self.connected:
+                self.log.debug("Waiting for next incoming data.")
+                await self.read_and_dispatch()
+        except Exception as e:
+            self.log.exception(f"read_loop failed. Disconnecting. {e!r}")
+            await self.close_client(cancel_read_loop_task=False)
+
+    @abc.abstractmethod
+    async def read_and_dispatch(self) -> None:
+        """Read, parse, and dispatch one item of data.
+
+        Subclasses need to implement this method such that it reads and parses
+        data and then dispatches handling the data to a method suitable for the
+        subclass. Methods that might be helpful include:
+
+        * `read_json` to read json-encoded data
+        * `read_str` to read terminated strings
+        * `read_into` to read binary structs
+        """
+        raise NotImplementedError()
+
+    async def close_client(self, cancel_read_loop_task: bool = True) -> None:
+        """Stop sending telemetry and close the client.
```

Comment:
Please change to something like "stop the read loop and close the client"

you might do a case-blind global search for "telemetry"